fontforge/tottfgpos.c

```diff
@@ -484,13 +484,23 @@ static void dumpcoveragetable(FILE *gpos,SplineChar **glyphs) {
     /* figure out whether it is better (smaller) to use an array of glyph ids */
     /*  or a set of glyph id ranges */
 
-    for ( i=0; glyphs[i]!=NULL; ++i ) {
-	if ( glyphs[i]->ttf_glyph<=last )
-	    IError("Glyphs must be ordered when creating coverage table");
-	if ( glyphs[i]->ttf_glyph!=last+1 )
-	    ++range_cnt;
-	last = glyphs[i]->ttf_glyph;
-    }
+	// We will not emit glyphs with -1 identifiers.
+	// We count the valid glyphs and the ranges.
+	int glyph_cnt = 0;
+	for (i=0; glyphs[i]!=NULL; i++) {
+		if (i > 0 && glyphs[i]->ttf_glyph <= glyphs[i-1]->ttf_glyph)
+			IError("Glyphs must be ordered when creating coverage table");
+		if (glyphs[i]->ttf_glyph < 0) {
+			fprintf(stderr, "-1 glyph index in dumpcoveragetable.\n");
```

Comment:
This is all true. I suppose LogError is appropriate here, even if it's not an error. I'll make the change when I add more comments and squash.